test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -892,6 +915,101 @@ def forward(self, x):
             node_list,
         )
 
+    def test_fold_quantize(self):
+        """Test to make sure the quantized model gets quantized weight (quantize op is folded)
+        """
+        m = self._get_pt2e_quantized_linear()
+        node_occurrence = {
+            # quantize op for weight node is folded
+            ns.call_function(torch.ops.quantized_decomposed.quantize_per_tensor.default): 2,
+            ns.call_function(torch.ops.quantized_decomposed.dequantize_per_tensor.default): 3,
+        }
+        self.checkGraphModuleNodes(m, expected_node_occurrence=node_occurrence)
+
+    def test_dont_fold_other_constant(self):
+        """Make sure the constant propagation does not apply to things unrelated to
+        quantization
+        """
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(2, 2)
+                self.dont_fold_me = torch.randn(2, 2)
+
+            def forward(self, x):
+                t = self.dont_fold_me.t()
+                return self.linear(x) + t
```

Comment:
You can just refactor this to take is non_const_fold op as arg, so that you can combine this with _get_pt2e_quantized_linear

Reply:
feels like this might make the original helper hard to read, so I'll keep this for now, maybe need to refactor the prepare -> calibrate -> convert code to a single call to deduplicate code

Reply:
I wouldnt imagine it will be too hard to read. I just want to reduce boilerplate stuff. But not too opinionated here.

Reply:
maybe I can factor out the quantization part of code